L46-core-tls-credential-API.md

```diff
@@ -0,0 +1,339 @@
+If that doesn't meet users' requirements, they could implement their own providers.
+
+Besides major credential reloading change, we are also planning to:
+1. make the server authorization original designed for client side only applicable to both the client side and the server side. 
+This is based on some recent user requirement, and it may also be used as the CRL check since CRL is not supported in gRPC core. 
+2. add the options to configure the min/max TLS version that will be used in handshake
+
+### Related Proposals:
+
+[L46: Create a new TLS credential API that supports SPIFFE mutual TLS](https://github.com/grpc/proposal/pull/98)
+
+## Proposal
+
+### Changes around TLS credentials and Options
+This part of the proposal introduces changes made to existing TLS credential APIs. Changes include:
+1. Replacing all the occurrence of "*_server_authorization_*" to "*_authorization_"
+2. Removing `grpc_tls_key_materials_config`, `grpc_tls_credential_reload_config` and `grpc_tls_credential_reload_arg` 
+3. Adding API `grpc_tls_credentials_options_set_certificate_provider`, `grpc_tls_server_credentials_set_min_tls_version` and `grpc_tls_server_credentials_set_max_tls_version`
+4. Adding API `grpc_tls_credentials_options_set_root_cert_name` and `grpc_tls_credentials_options_set_identity_cert_name`
+
+Here is how the API would look like after these changes. For simplicity, all the comments are omitted.
+
+```c
+/** --- TLS channel/server credentials ---
+ * It is used for experimental purpose for now and subject to change. */
+
+typedef struct grpc_tls_error_details grpc_tls_error_details;
+
+typedef struct grpc_tls_authorization_check_config grpc_tls_authorization_check_config;
+
+typedef struct grpc_tls_credentials_options grpc_tls_credentials_options;
+
+GRPCAPI grpc_tls_credentials_options* grpc_tls_credentials_options_create(void);
+
+GRPCAPI int grpc_tls_credentials_options_set_cert_request_type(
+    grpc_tls_credentials_options* options,
+    grpc_ssl_client_certificate_request_type type);
+
+GRPCAPI int grpc_tls_credentials_options_set_server_verification_option(
+    grpc_tls_credentials_options* options,
+    grpc_tls_server_verification_option server_verification_option);
+
+GRPCAPI int grpc_tls_credentials_options_set_authorization_check_config(
+    grpc_tls_credentials_options* options,
+    grpc_tls_authorization_check_config* config);
+
+GRPCAPI int grpc_tls_credentials_options_set_root_cert_name(
+    grpc_tls_credentials_options* options,
+    const char* root_cert_name);
+
+GRPCAPI int grpc_tls_credentials_options_set_identity_cert_name(
+    grpc_tls_credentials_options* options,
+    const char* identity_cert_name);
+
+/** Sets the credential provider. */
+GRPCAPI int grpc_tls_credentials_options_set_certificate_provider(
+    grpc_tls_credentials_options* options,
+    grpc_tls_certificate_provider* provider);
+
+/** Sets the minimum TLS version that will be negotiated during the TLS
```

Comment:
Please comment that if this function is not called, then the underlying SSL library will pick min and max TLS version automatically.